next/components/pagination.tsx

```diff
@@ -0,0 +1,126 @@
+import Link from "next/link";
+import { useRouter } from "next/router";
+import { useTranslation } from "next-i18next";
+import { useState } from "react";
+
+import Arrow from "@/styles/icons/arrow-down.svg";
+
+import { Button } from "@/wunder-component-library/button";
+
+export type PaginationProps = {
+  currentPage?: number;
+  totalPages?: number;
+  prevEnabled?: boolean;
+  nextEnabled?: boolean;
+  setNextPage?: () => void;
+  setPrevPage?: () => void;
+  prevPageHref?: string;
+  nextPageHref?: string;
+};
+
+type PaginationComponentProps = {
+  paginationProps: PaginationProps;
+  focusRestoreRef?: React.RefObject<HTMLDivElement>;
+};
+
+const MaybeLink = ({ href, children }) =>
+  href ? (
+    <Link passHref scroll={false} href={href}>
+      {children}
+    </Link>
+  ) : (
+    children
+  );
+
+export function Pagination({
+  paginationProps = {},
+  focusRestoreRef,
+  ...props
+}: PaginationComponentProps) {
+  const {
+    currentPage,
+    totalPages,
+    nextEnabled = true,
+    prevEnabled = true,
+    setNextPage,
+    setPrevPage,
+    prevPageHref,
+    nextPageHref,
+  } = paginationProps;
+  const { t } = useTranslation();
+
+  const router = useRouter();
+  const [isLoading, setIsLoading] = useState<"forward" | "back" | false>(false);
+  const numbers = [currentPage, totalPages].filter((n) => !isNaN(n));
+
+  const restoreScroll = () => {
+    focusRestoreRef?.current.scrollIntoView({
+      behavior: "smooth",
+      block: "start",
+    });
+    const focusable = focusRestoreRef?.current.querySelector(
+      'button, [href], input, select, textarea, [tabindex]:not([tabindex="-1"])'
+    );
+    if (focusable) {
+      (focusable as HTMLElement).focus({ preventScroll: true });
+    }
+  };
+
+  const handlePrevClick = (e) => {
```

Comment:
If you wan't to be extra nice for the user, these click handlers could check if user intended to open the link in new tab with for example cmd+click and then just let the click go to the `<Link>`

Reply:
naturally you would need to also change from `<button>` to `<Link>`